grpc_publisher_service.py

```diff
@@ -513,6 +514,83 @@ def get_platform_info(platform_letter):
             return platform_info, None
+        
+        def handle_test_client(
+            msisdn,
+            payload,
+            context,
+            response,
+        ):
+          
+            try:
+                sms_sent_time_epoch = payload.get("time")
+                test_id = payload.get("id")
+
+                try:
+                    test_id = int(test_id)
+                    logger.debug(f"Converted test_id to integer: {test_id}")
+                except ValueError:
+                    return self.handle_create_grpc_error_response(
+                        context,
+                        response,
+                        f"Invalid test_id: {test_id}. Must be an integer.",
+                        grpc.StatusCode.INVALID_ARGUMENT,
+                        error_prefix="Invalid Payload",
+                        send_to_sentry=True,
+                    )
+
+                if not sms_sent_time_epoch or not test_id:
+                    return self.handle_create_grpc_error_response(
+                        context,
+                        response,
+                        "Invalid payload: Missing required fields (time or id).",
+                        grpc.StatusCode.INVALID_ARGUMENT,
+                        error_prefix="Invalid Payload",
+                        send_to_sentry=True,
+                    )
+
+                try:
+                    sms_sent_time = datetime.datetime.fromtimestamp(int(sms_sent_time_epoch))
+                    logger.debug(f"Converted sms_sent_time: {sms_sent_time}, type: {type(sms_sent_time)}")
+                except (ValueError, OSError) as e:
+                    return self.handle_create_grpc_error_response(
+                        context,
+                        response,
+                        f"Invalid epoch time: {sms_sent_time_epoch}. Must be a valid Unix timestamp.",
+                        grpc.StatusCode.INVALID_ARGUMENT,
+                        error_prefix="Invalid Payload",
+                        send_to_sentry=True,
+                    )
+
+                test_client = TestClient()
+                _, test_error = test_client.update_test_message(test_id=test_id, sms_sent_time=sms_sent_time)
+
+                if test_error:
+                    return self.handle_create_grpc_error_response(
+                        context,
+                        response,
+                        test_error,
+                        grpc.StatusCode.NOT_FOUND if "not found" in test_error else grpc.StatusCode.INTERNAL,
+                        error_prefix="Error Updating Test Message",
+                        send_to_sentry=True,
+                    )
+
+                return response(
+                    message="Successfully updated test message in the database",
+                    publisher_response="Successfully published message to Test Plaftform",
+                    success=True,
+                )
+
+            except Exception as exc:
```

Comment:
no need for the try-except block the caller function will handle any uncaught exception